### PR TITLE
Adds Add idea page and event and idea functions

### LINKED ISSUE #60: Make new idea - frontend component

### PR DESCRIPTION
**Pull request by jonathandavidlewis**
closes #60 

Ideas can be created, 
promoted to events, 
Events can be demoted to ideas
Events can be deleted
Ideas can be deleted.
ideas and itineraries view populate correctly.

### CODE REVIEW

File: client/src/app.routing.js [hunk 1]
```diff
@@ -50,7 +50,10 @@ const routing = function ($stateProvider, $urlRouterProvider, $locationProvider)
     url: '/plan/{planId}',
     component: 'plan',
     resolve: {
-      protect: redirectIfNotAuthed
+      //todo: Check if safe for minification
+      events: function($stateParams, EventService) {
```
Comment (AtreidesEnd) [inline client/src/app.routing.js]: won't be, make a helper below so you can do .$inject on it

File: client/src/modules/home/plan-card/plan-card.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import './plan-card.css';
 
 class PlanCardController {
   constructor() {
-    this. handleDeleteClick = () => {
+    this.handleDeleteClick = () => {
```
Comment (AtreidesEnd) [inline client/src/modules/home/plan-card/plan-card.js]: can we use the same syntax model you used in ideas-card here? declare outside constructor and bind inside

File: client/src/modules/plan/planner/ideas/ideas.js [hunk 3]
```diff
@@ -12,10 +14,33 @@ import './ideas.css';
...
+    this.promoteEvent = this.promoteEvent.bind(this);
+    this.$onInit = this.$onInit.bind(this);
+  }
+
+  loadEvents(events) {
+    this.events = events.filter((event) => event.status === 'idea');
+  }
+
+  deleteEvent(eventId) {
+    this.EventService.deleteEvent(eventId).then(
```
Comment (AtreidesEnd) [inline client/src/modules/plan/planner/ideas/ideas.js]: are you missing () => { . here?

File: client/src/modules/plan/planner/ideas/ideas.js [hunk 4]
```diff
@@ -12,10 +14,33 @@ import './ideas.css';
...
+      this.EventService.loadEventsByPlanId(this.$stateParams.planId).then((events) => {
+        this.loadEvents(events);
+      })
```
Comment (AtreidesEnd) [inline client/src/modules/plan/planner/ideas/ideas.js]: semicolon

File: client/src/modules/plan/planner/ideas/ideas.js [hunk 5]
```diff
@@ -12,10 +14,33 @@ import './ideas.css';
 
 class IdeasController {
   constructor(EventService, $stateParams) {
+    this.EventService = EventService;
     this.title = 'This is the ideas component';
```
Comment (AtreidesEnd) [inline client/src/modules/plan/planner/ideas/ideas.js]: this isn't needed anymore is it?

File: client/src/modules/plan/planner/ideas/new-idea-button/new-idea-button.css [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.new-plan-button {
+  width: 100%;
+  padding: 10px;
+  background-color: var(--b5);
```
Comment (AtreidesEnd) [inline client/src/modules/plan/planner/ideas/new-idea-button/new-idea-button.css]: what's --b5?

File: client/src/modules/plan/planner/itinerary/itinerary-card/itinerary-card.html [hunk 7]
```diff
@@ -7,6 +7,8 @@
         <div class="itinerary-card-startTime"><b>Starts: </b><span>{{$ctrl.event.startTime}}</span></div>
         <div ng-if="$ctrl.event.endTime" class="itinerary-card-endTime"><b>Ends: </b><span>{{$ctrl.event.endTime}}</span></div>
         <div class="itinerary-card-description"><b>Details: </b><span>{{$ctrl.event.description}}</span></div>
+        <spaan ><a ng-click="$ctrl.handleDemoteClick()">DEMOTE</a></spaan>
```
Comment (AtreidesEnd) [inline client/src/modules/plan/planner/itinerary/itinerary-card/itinerary-card.html]: typo - span

File: client/src/modules/plan/planner/make-idea/make-idea.html [hunk 8]
```diff
@@ -0,0 +1,31 @@
+<div class="make-idea-container flex-col flex-start flex-align-stretch">
+  <div class="make-idea-header"><span>Awesome! <br /> Let's get started</span></div>
+  <div class="make-idea-form-container">
+    <form ng-submit="$ctrl.submit()">
+      <div class="make-idea-input-block">
+        <label for="idea-title">Give your idea a name:</label>
+        <input class="awgl-form-input form-control" id="idea-title" ng-model="$ctrl.title"
```
Comment (AtreidesEnd) [inline client/src/modules/plan/planner/make-idea/make-idea.html]: switch from using id's to using names and give the overall form a name, you can reference things within a form more easily when validation gets used with formName.fieldName

**Comment by jonathandavidlewis** [comment]
Feedback implimented. 
Also, updated buttons look:
image

**Pull request merged**